src/helper/ssg/index.ts

```diff
@@ -19,24 +20,90 @@ const generateFilePath = (routePath: string, outDir: string) => {
   return path.join(outDir, fileName)
 }
 
+interface SSGParam {
+  [key: string]: string
+}
+type SSGParams = SSGParam[]
+interface SSGParamsMiddleware {
+  (generateParams: () => SSGParams | Promise<SSGParams>): MiddlewareHandler
```

Comment:
How about passing an instance of Context?

```ts
(generateParams: (c: Context) => SSGParams | Promise<SSGParams>): MiddlewareHandler
```

By doing this, you can write cool code like the following:

```ts
app.get(
  '/shops',
  createMiddleware<{ Variables: { shops: Shop[] } }>(async (c) => {
    const shops = await getShops()
    c.set('shops', shops)
  }),
  ssgParams(async (c) => {
    return c.var.shops.map((shop) => ({ id: shop.id }))
  }),
  async (c) => {
    return c.render(
      <div>
        <h1>ラーメン屋さん</h1>
        <ul>
          {c.var.shops.map((shop) => {
            return (
              <li>
                <a href={`/shops/${shop.id}`}>{shop.name}</a>
              </li>
            )
          })}
        </ul>
      </div>
    )
  }
)
```

Reply:
It's really cool idea!
I'll implement it.

Reply:
Thank you. We might need to change the type, but I'll do it after this PR is merged!